src/config.js

```diff
@@ -27,8 +36,37 @@ const Config = CoreObject.extend({
   },
 
   /**
-   * The apiKey for authenticating against shopify. This is your api client's
-   * public api token. Not the shared secret. Set during initialation.
+   * An object with keys for deprecated properties and values as functions that
+   * will transform the value into a usable value.
+   * @attribute deprecatedProperties
+   * @default { myShopifyDomain: this.transformMyShopifyDomain }
+   * @type Object
+   * @private
+   */
+  deprecatedProperties: {
+    myShopifyDomain: 'transformMyShopifyDomain'
+  },
+
+  /**
+   * Transform the myShopifyDomain config to a domain config.
+   * @method transformMyShopifyDomain
+   * @static
+   * @private
+   * @param {String} subdomain The original subdomain on myshopify.com
+   * @param {Object} attrs. The config attributes to be transformed to a
+   * non-deprecated state.
+   * @return {Object} the transformed config attributes.
+   */
+  transformMyShopifyDomain(subdomain, attrs) {
+    logger.warn('Config - ',
+       `myShopifyDomain is deprecated,
+       please use domain and provide the whole myshopify.com domain.`);
```

Comment:
`provide the full shop domain`. The point of this is that the merchant may wish to use their own CNAME, and also, with products like sello: we're introducing shops that are subdomains of other top level shopify domains.